### PR TITLE
This commit will: Task 07-> Implement POST to /api/articles/:article_…

### PR DESCRIPTION
**Pull request by edhawx**
…id/comments with a 201 status code, plus implement testing on this endpoint, responding with 400 and 404

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
...
+        expect(body.msg).toBe("400 - Bad request");
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Responds with 201 code and the posted comment", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
```
Comment (MKD01) [inline __tests__/app.test.js:189]: Love the comment body 😂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
...
+      .then((res) => {
+        expect(res.body.comment).toEqual(
+          expect.objectContaining({
+            body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+            votes: 0,
+            author: "rogersop",
+            article_id: 1,
+            created_at: expect.any(String),
+          })
+        );
```
Comment (MKD01) [inline __tests__/app.test.js:204]: Really good expect statement, but this is missing the comment_id which is very important and something we should expect to receive.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
...
+      body: "test",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("404 - Not found, user doesn't exist!");
+      });
+  });
```
Comment (MKD01) [inline __tests__/app.test.js:220]: Really good error handling test.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .post(`/api/articles/1/comments`)
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe(
+          `400 - Bad request, you haven't typed a comment!`
+        );
+      });
+  });
```
Comment (MKD01) [inline __tests__/app.test.js:284]: This is a nice and interesting edge case

File: __tests__/app.test.js [hunk 5]
```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
...
 });
```
Comment (MKD01) [inline __tests__/app.test.js:285]: Some extra tests to consider would be:

What should happen when the newComment doesnt not have a body property at all?
What should happen when the newComment has extra key value pairs?

**Review by MKD01 — COMMENTED** [review]
Really good work overall, but there are a couple extra test cases to consider. Please do add those in but once you have you can merge that in without needing to send us an updated pr
